### ISSUE TITLE
"Error: Authentication Failure (HTTP Error 404: Not Found)" when attempting to sign in

### ISSUE THREAD

**Issue by jamesderlin** [smkent/safeway-coupons#112]
Starting a few days ago, I've started getting the following error when attempting to run `safeway-coupons`:

```
Clipping coupons for Safeway account REDACTED
Would send email to REDACTED
>>>>>>
Safeway account: REDACTED
Error: Authentication Failure (HTTP Error 404: Not Found)
<<<<<<
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/session.py", lin
e 58, in __init__
    self._login(account)   
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/session.py", lin
e 97, in _login
    with self._chrome_driver() as driver:
  File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)  
           ^^^^^^^^^^^^^^  
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/session.py", lin
e 71, in _chrome_driver
    with chrome_driver(headless=headless) as driver:
  File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/chrome_driver.py", line 38, in chrome_driver
    driver = uc.Chrome(options=options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/undetected_chromedriver/__init__.py", line 258, in __init__
    self.patcher.auto()
  File "/usr/local/lib/python3.12/site-packages/undetected_chromedriver/patcher.py", line 178, in auto
    self.unzip_package(self.fetch_package())
                       ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/undetected_chromedriver/patcher.py", line 287, in fetch_package
    return urlretrieve(download_url)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 240, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
                            ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)   
             ^^^^^^^^^^^   
  File "/usr/local/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/safeway-coupons", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/app.py", line 125, in main
    sc.clip_for_account(account)
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/safeway.py", line 38, in clip_for_account  
    swy = SafewayClient(account, self.debug_dir)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/client.py", line 17, in __init__
    self.session = LoginSession(account, debug_dir)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/safeway_coupons/session.py", line 66, in __init__
    raise AuthenticationFailure(e, account) from e
safeway_coupons.errors.AuthenticationFailure: Authentication Failure (HTTP Error 404: Not Found)
```

Printing the `url` member for the thrown `urllib.error.HTTPError` indicates that the failed URL is https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/122.0.6261.57/linux64/chromedriver-linux64.zip.  I'm not sure if this is some temporary failure, if the chromedriver needs to be updated, or what.

**Comment by datslo** [comment]
I'm getting the 404 as well

**Comment by jvasa** [comment]
I'm seeing this too. Thanks

**Comment by Juberstine** [comment]
Same issue

**Comment by datslo** [comment]
@jamesderlin just wanted to make sure you're aware of this issue. any solutions in mind?

**Comment by smkent** [comment]
Thanks everyone for the report. It seems undetected_chromedriver attempts to re-download the chrome driver binary on every invocation, and the version it was trying to download is either out of date or no longer exists.

I bumped the build (https://github.com/smkent/safeway-coupons/pull/113) and generated a new container image. safeway-coupons appears working for me again. If things still aren't working with the new container, please (re)-open an issue. Thanks!

**Comment by sergberg5** [comment]
I pulled down the app brand new on my local machine and worked out of the box with the cli command. I did this because my deployed version on a cloud provider (custom because of the impreva thing a while back) version was throwing the above error. I was checking my dependencies and was lazy so I upgraded all at once to get it going and one of the below must have been it for me. My chrome versions were identical between the two machines. ¯\_(ツ)_/¯ I hope this isn't a red herring, if so, im sorry

Successfully installed charset-normalizer-3.3.2 dataclasses-json-0.6.4 requests-2.31.0 selenium-4.18.1 typing-extensions-4.10.0 undetected-chromedriver-3.5.5
